src/main/java/com/suse/saltstack/netapi/datatypes/cherrypy/Stats.java

```diff
@@ -0,0 +1,20 @@
+package com.suse.saltstack.netapi.datatypes.cherrypy;
+
+public class Stats {
+
+    private final Applications applications;
+    private final HttpServer httpServer;
+
+    public Stats(Applications applications, HttpServer httpServer) {
+        this.applications = applications;
+        this.httpServer = httpServer;
```

Comment:
Looks like here it should check if apps and server are not null, throw the exception otherwise. This will save you extra code everywhere to check if the objects are there as well as not to initiate `Stats` class with `null`, like you do below in `StatsDeserializer`.